### ISSUE TITLE
abstained label in embedding approach

### ISSUE THREAD

**Issue by BUGLI27** [BunsenFeng/AbstainQA#1]
Hi, great work, but i'm a little confused about the abstained label in approach-embedding.
During the phase one, when model can correctly answer the question, flag is 1, otherwise 0.
```python
if lm_utils.answer_parsing(response) == d["answer"]:
      correct_flags.append(1)
  else:
      correct_flags.append(0)
```
During the phase two, linear_model is trained by the correct_flags_dev.
```python
linear_model = train_linear_model(linear_model, dev_embeddings, torch.tensor(correct_flags_dev))
```
Intuitively, when the `torch.argmax(outputs, dim=1)` is 0, means the model can't correctly answer the question, and should abstain.
However, in the `metrics.py`, it seems that when `abstain_flags[i] == 1` means abstained.
```python
# group A: answered and correct
    # group B: abstained and correct
    # group C: answered and incorrect
    # group D: abstained and incorrect
    A = 0
    B = 0
    C = 0
    D = 0
    for i in range(len(correct_flags)):
        if abstain_flags[i]:
            if correct_flags[i]:
                B += 1
            else:
                D += 1
        else:
            if correct_flags[i]:
                A += 1
            else:
                C += 1
```
Did I make a mistake somewhere?

**Comment by BunsenFeng** [comment]
Hi BUGLI27, thank you for spotting this! Yes the output of torch.argmax(outputs, dim=1) should be flipped: I'm temporarily doing a quick fix by changing line 154 from abstain_flags.append(abstain_flag.item()) to abstain_flags.append(1-abstain_flag.item()). The abstain_score metric should be alright as `outputs[0][0].item()` is the probability of abstaining.

We flipped the definition of 0s and 1s during the project and I suspect this is a versioning issue in the release code. Thank you for spotting this and let me know if you find any additional 0/1 inconsistencies.
